src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        switch (random.nextInt(5)) {
```

Comment:
Consider defining a constant for the number of figure types instead of using the magic number `5`. This will improve code readability and maintainability. For example, `private static final int FIGURE_TYPES_COUNT = 5;` and use it in `random.nextInt(FIGURE_TYPES_COUNT)`.